src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -587,7 +612,17 @@ HRESULT CProfilerManager::AddInstrumentationMethod(
         return E_OUTOFMEMORY;
     }
 
-    return this->AddInstrumentationMethod(pInstrumentationMethod.release(), ppInstrumentationMethod);
+    CComPtr<CConfigurationSource> pSource;
```

Comment:
>pSource [](start = 34, length = 7)

Currently, this CConfigurationSource will always return an empty IEnumInstrumentationMethodSettings. Would it make more sense to just pass NULL to the following AddInstrumentationMethod()?

Reply:
I think it's better as-is so that downstream doesn't have to account for nullptr. All of the SAL annotations say that it will not be nullptr ( _In_ ).

Reply:
Sure, not a nullptr, but why call EnumSettings from a newly created ConfigurationSource (which won't have any settings)?

Reply:
Because it will return an object that has nothing to enumerate so that I can pass in something that is not nullptr. And I don't want to just new up a concrete instance of IEnumInstrumentationMethodSettings and pass that in because that would require me to create a vector<CComPtr<IInstrumentationMethodSetting>>, make an enumerator from that, and then pass that in. It's just as much code as what I have already, but introduces implementation detail to the ProfilerManager that it doesn't really care to know about. It is primarily concerned about creating sources and passing them through for use elsewhere.

Reply:
Thanks for the explanation!